solidity/contracts/bridge/library/FraudVerification.sol

```diff
@@ -0,0 +1,285 @@
+// SPDX-License-Identifier: MIT
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity 0.8.4;
+
+import {BytesLib} from "@keep-network/bitcoin-spv-sol/contracts/BytesLib.sol";
+import {BTCUtils} from "@keep-network/bitcoin-spv-sol/contracts/BTCUtils.sol";
+import {
+    CheckBitcoinSigs
+} from "@keep-network/bitcoin-spv-sol/contracts/CheckBitcoinSigs.sol";
+import "../Bridge.sol";
+
+library FraudVerification {
+    using BytesLib for bytes;
+    using BTCUtils for bytes;
+    using BTCUtils for uint32;
+
+    struct Parameters {
+        /// The amount of ETH the party challenging the wallet for fraud needs
+        /// to deposit.
+        uint256 fraudChallengeDepositAmount; //TODO: Initialize
+    }
+
+    struct FraudChallenge {
+        address challenger;
+        uint256 ethDepositAmount;
+        uint32 reportedAt;
+        bool defended;
+    }
+
+    struct Data {
+        Parameters parameters;
+        /// Collection of all submitted fraud challenges indexed by challenge
+        /// key built as keccak256(walletPublicKey|sighash|v|r|s).
+        mapping(uint256 => FraudChallenge) fraudChallenges;
```

Comment:
Maybe just `challenges`? We are already in the "fraud" context.

Reply:
Done.